IEx uses erlang shell rendering for erlang docs

sabiwara opened the pull request:
As suggested [here](https://github.com/elixir-lang/elixir/pull/12802#issuecomment-1637849231), use [shell_docs:render/3](https://www.erlang.org/doc/man/shell_docs.html#render-2) to render Erlang docs in IEx:

<img width="716" alt="Screenshot 2023-07-17 at 21 09 58" src="https://github.com/elixir-lang/elixir/assets/11598866/4e028a45-595c-46d5-9817-d75ea64173a7">

josevalim commented:
I think there is a bunch of code in IO.ANSI.Docs you can now remove, as it no longer deals with Erlang AST. It may be worth doing a pass on the code to find other areas that can now be removed (perhaps blame when the feature was added and see related areas?).

sabiwara commented:
Sounds good, will do !

sabiwara commented:
> I think there is a bunch of code in IO.ANSI.Docs you can now remove, as it no longer deals with Erlang AST. It may be worth doing a pass on the code to find other areas that can now be removed (perhaps blame when the feature was added and see related areas?).

OK, so before we can clean up we'll also need to migrate the module doc and type docs (`t`) to use `:shell_docs`.
I think it makes sense to migrate thoroughly, just wanted to confirm before I move forward.

<img width="877" alt="Screenshot 2023-07-18 at 8 19 03" src="https://github.com/elixir-lang/elixir/assets/11598866/df83c22c-9bfc-4240-8d4e-becc4d142443">

----

<img width="464" alt="Screenshot 2023-07-18 at 8 17 56" src="https://github.com/elixir-lang/elixir/assets/11598866/22082501-c079-4754-91d1-386271a1de7c">

josevalim commented:
Correct. We can print all Erlang entries with Erlang tools.

sabiwara commented:
@josevalim I think we should be good:

- added support for erlang module doc
- added support for erlang types
- cleaned up the ansi docs erlang support added in https://github.com/elixir-lang/elixir/pull/9843/files

<img width="920" alt="Screenshot 2023-07-19 at 23 07 57" src="https://github.com/elixir-lang/elixir/assets/11598866/9b713936-4d5a-41d3-a62a-a94f7d75eb73">

<img width="324" alt="Screenshot 2023-07-19 at 23 08 26" src="https://github.com/elixir-lang/elixir/assets/11598866/ae8beff0-39a9-4400-bbaf-ac08aeb45133">

sabiwara commented on lib/iex/lib/iex/introspection.ex:
```diff
@@ -659,68 +666,86 @@ defmodule IEx.Introspection do
   Prints the types for the given module and type documentation.
   """
   def t(module) when is_atom(module) do
-    case Typespec.fetch_types(module) do
-      :error ->
-        no_beam(module)
+    case :code.get_doc(module) do
+      {:ok, {:docs_v1, _, :erlang, _, _, _, _} = erl_docs} ->
+        :shell_docs.render_type(module, erl_docs) |> IO.puts()
```
Note: I've tried to refactor these  three `t` clauses but each implementation is slightly different  in a way that makes it quite difficult to abstract, so I ended up just nesting the whole implementation in an extra case.
